include/mesh.h

```diff
@@ -30,11 +30,11 @@ class Mesh {
   virtual void read_mesh(const std::string& filename) = 0;
 
   //! Compute material point location
-  virtual void compute_material_points(unsigned ngauss_points) = 0;
+  virtual void generate_material_points(unsigned ngauss_points) = 0;
 
   //! Get material properties from json object
   void assign_material_properties(
-      const std::shared_ptr<MaterialProperties>& material);
+      std::vector<std::unique_ptr<MaterialProperties>> material);
```

Comment:
If this is receiving material properties from JSON object, this should get a JSON object

Reply:
We can call constructor of `MaterialProperties` with input of the JSON object

Reply:
Another way is we can refactor them and just output and input JSON object, but make `MaterialProperties` in `mesh`. Would that be cleaner?